roofit/histfactory/src/HistFactoryNavigation.cxx

```diff
@@ -1127,9 +1095,10 @@ namespace RooStats {
 
       /////// NODE SIZE
       {
-   TIter itr = components.createIterator();
-   RooAbsArg* arg = NULL;
-   while( (arg=(RooAbsArg*)itr.Next()) ) {
+   //TIter itr = components.createIterator();
+   //RooAbsArg* arg = NULL;
+   //while( (arg=(RooAbsArg*)itr.Next()) ) 
+   for (auto *arg : dynamic_range_cast<RooAbsArg *>(components)) {
```

Comment:
```suggestion
   for (auto * component : static_range_cast<RooAbsReal *>(components)) {
```
Remove commented out lines, and you can directly cast to `RooAbsReal` which is the type required in this loop (you can delete the line below then). Also it can be a `static_cast` because the result is never checked for `nullptr` anyway.